overlord/snapstate/backend/export_test.go

```diff
@@ -43,3 +45,11 @@ func MockCommandFromSystemSnap(f func(string, ...string) (*exec.Cmd, error)) (re
 		commandFromSystemSnap = old
 	}
 }
+
+func MockSystemdNew(f func(systemd.InstanceMode, systemd.Reporter) systemd.Systemd) (restore func()) {
```

Comment:
after further thinking I'm not sure we want to switch exactly to this kind of approach. The problem is that the systemd package has 3 ways to build Systemd implementations, and some packages use all of them (configcore is a prime example), and switching to this there would be confusing.